### PR TITLE
Only remove extra archives

### PR DESCRIPTION
**Pull request by doomspork**
Alright @ericmj, here's another go at #3345.  Let me know your thoughts.

### CODE REVIEW

File: lib/mix/lib/mix/tasks/archive.install.ex [hunk 1]
```diff
@@ -64,6 +62,8 @@ defmodule Mix.Tasks.Archive.Install do
       if Mix.Utils.copy_path!(src, archive, opts) do
         Mix.shell.info [:green, "* creating ", :reset, Path.relative_to_cwd(archive)]
         Mix.Local.check_archive_elixir_version archive
+
+        unless Enum.member?(previous, archive), do: remove_previous_versions(previous)
```
Comment (ericmj) [inline lib/mix/lib/mix/tasks/archive.install.ex]: You can do `archive in previous` instead.

Comment (doomspork) [inline lib/mix/lib/mix/tasks/archive.install.ex]: Awesome, thanks for the tip.  I'll go ahead and make that change.

**Comment by doomspork** [comment]
@ericmj updated with your suggestion, anything else?

**Comment by ericmj** [comment]
Thanks @doomspork!

**Pull request merged**